modules/tensor_mechanics/src/kernels/StressDivergenceTensors.C

```diff
@@ -79,13 +84,19 @@ StressDivergenceTensors::StressDivergenceTensors(const InputParameters & paramet
 Real
 StressDivergenceTensors::computeQpResidual()
 {
-  return _stress[_qp].row(_component) * _grad_test[_i][_qp];
+  if ((_dt > 0) && ((_alpha != 0) || (_zeta != 0)))
```

Comment:
Stuff like this needs to have comments! Also the alpha!=0 || zeta!= 0 can be folded into a bool flag with a _meaningful_ name at construction time.
Now that I think about this, don't do any of this in ```StressDivergenceTensors```! Create a derived class. We do not want our classes to grow to become monolithic multipurpose monsters like in certain other mechanics systems ;-).
This is a straight forward case wher yuor derived class can add a few variables and overwrite the computeQpResidual et al. methods.